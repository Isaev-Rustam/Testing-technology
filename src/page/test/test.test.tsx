import { describe, vi } from 'vitest';

import { square } from './test';

describe('Base Test', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('square', () => {
    const spy = vi.spyOn(Math, 'sqrt');
    expect(square(3)).toBe(false);
    expect(spy).not.toHaveBeenCalled();
  });

  it('square', () => {
    const spy = vi.spyOn(Math, 'sqrt');
    expect(square(4)).toBe(2);
    expect(spy).toHaveBeenCalled();
    expect(square(4)).toMatchSnapshot();
  });
});
