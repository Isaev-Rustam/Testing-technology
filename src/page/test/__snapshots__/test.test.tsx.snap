// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Base Test > square 1`] = `2`;
